### PR TITLE
Simplify the TablesRecordedTestSanitizer to employ regex

### LINKED ISSUE #11778: Simplify the SanitizeUri method in TablesRecordedTestSanitizer
Could be simplified with a regex replace for the `sig` parameter.

### PR DESCRIPTION
**Pull request by christothes**
Fixes #11778

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/tests/TablesRecordedTestSanitizer.cs [hunk 1]
```diff
@@ -21,19 +22,13 @@ public override string SanitizeVariable(string variableName, string environmentV
...
-        public string SanitizeQueryParameters(string uri)
-        {
-            var builder = new UriBuilder(base.SanitizeUri(uri));
-            var query = new UriQueryParamsCollection(builder.Query);
-            if (query.ContainsKey(SignatureQueryName))
+            if (SignatureRegEx.IsMatch(uri))
             {
-                query[SignatureQueryName] = SanitizeValue;
-                builder.Query = query.ToString();
+                var match = SignatureRegEx.Replace(uri, $"$1{SanitizeValue}");
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/tests/TablesRecordedTestSanitizer.cs:27]: I think you don't need IsMatch check. `SignatureRegEx.Replace` should do it.

File: sdk/tables/Azure.Data.Tables/tests/TablesRecordedTestSanitizer.cs [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
 using System;
+using System.Text.RegularExpressions;
 using Azure.Core.TestFramework;
 
 namespace Azure.Data.Tables.Tests
 {
     public class TablesRecordedTestSanitizer : RecordedTestSanitizer
     {
-        private const string SignatureQueryName = "sig";
+        private Regex SignatureRegEx = new Regex(@"([\x0026|&|?]sig=)([\w\d%]+)", RegexOptions.Compiled);
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/tests/TablesRecordedTestSanitizer.cs:12]:
```suggestion
        private Regex SignatureRegEx = new Regex(@"([^\w]sig=)([\w\d%]+)", RegexOptions.Compiled);
```
??

Comment (christothes) [reply sdk/tables/Azure.Data.Tables/tests/TablesRecordedTestSanitizer.cs:12]: I wanted to account for if the uri was already encoding unicode ampersand as well (\u0026). Perhaps that is overkilll?

Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/tests/TablesRecordedTestSanitizer.cs:12]: I'm fine with either option.

**Review by pakrym — APPROVED** [review]

**Pull request merged**